zerver/tests/test_rocketchat_importer.py

```diff
@@ -0,0 +1,672 @@
+        # Teams are a subset of rooms.
+        self.assert_length(room_id_to_room_map, 6)
+        self.assert_length(team_id_to_team_map, 1)
+        self.assert_length(dsc_id_to_dsc_map, 4)
+        self.assert_length(direct_id_to_direct_map, 2)
+
+        room_id = rocketchat_data["room"][0]["_id"]
+        self.assertIn(room_id, room_id_to_room_map)
+        self.assertEqual(room_id_to_room_map[room_id], rocketchat_data["room"][0])
+
+        team_id = rocketchat_data["room"][3]["teamId"]
+        self.assertIn(team_id, team_id_to_team_map)
+        self.assertEqual(team_id_to_team_map[team_id], rocketchat_data["room"][3])
+
+        dsc_id = rocketchat_data["room"][7]["_id"]
+        self.assertIn(dsc_id, dsc_id_to_dsc_map)
+        self.assertEqual(dsc_id_to_dsc_map[dsc_id], rocketchat_data["room"][7])
+
+        direct_id = rocketchat_data["room"][4]["_id"]
+        self.assertIn(direct_id, direct_id_to_direct_map)
+        self.assertEqual(direct_id_to_direct_map[direct_id], rocketchat_data["room"][4])
+
+    def test_convert_channel_data(self) -> None:
+        fixture_dir_name = self.fixture_file_name("", "rocketchat_fixtures")
+        rocketchat_data = rocketchat_data_to_dict(fixture_dir_name)
+
+        realm_id = 3
+        stream_id_mapper = IdMapper()
+
+        room_id_to_room_map: Dict[str, Dict[str, Any]] = {}
+        team_id_to_team_map: Dict[str, Dict[str, Any]] = {}
+        dsc_id_to_dsc_map: Dict[str, Dict[str, Any]] = {}
+        direct_id_to_direct_map: Dict[str, Dict[str, Any]] = {}
+
+        categorize_channels_and_map_with_id(
+            channel_data=rocketchat_data["room"],
+            room_id_to_room_map=room_id_to_room_map,
+            team_id_to_team_map=team_id_to_team_map,
+            dsc_id_to_dsc_map=dsc_id_to_dsc_map,
+            direct_id_to_direct_map=direct_id_to_direct_map,
+        )
+
+        zerver_stream = convert_channel_data(
+            room_id_to_room_map=room_id_to_room_map,
+            team_id_to_team_map=team_id_to_team_map,
+            stream_id_mapper=stream_id_mapper,
+            realm_id=realm_id,
+        )
+
+        # Only rooms are converted to streams.
+        self.assert_length(room_id_to_room_map, 6)
+        self.assert_length(zerver_stream, 6)
+
+        # Normal public stream
+        self.assertEqual(zerver_stream[0]["name"], "general")
+        self.assertEqual(zerver_stream[0]["invite_only"], False)
+        self.assertEqual(zerver_stream[0]["description"], "This is a general channel.")
+        self.assertEqual(zerver_stream[0]["rendered_description"], "")
+        self.assertEqual(zerver_stream[0]["realm"], 3)
+
+        # Private stream
+        self.assertEqual(zerver_stream[1]["name"], "random")
+        self.assertEqual(zerver_stream[1]["invite_only"], True)
+        self.assertEqual(zerver_stream[1]["description"], "")
+        self.assertEqual(zerver_stream[1]["rendered_description"], "")
+        self.assertEqual(zerver_stream[1]["realm"], 3)
```

Comment:
Let's use `realm_id` rather than `3` for these checks; that's a lot more readable.